### PR TITLE
gh-108682: [Enum] raise TypeError if super().__new__ called in custom __new__

### PR DESCRIPTION
**Pull request by ethanfurman**
----
:books: Documentation preview :books:: https://cpython-previews--108704.org.readthedocs.build/

* Issue: gh-108682

### CODE REVIEW

File: Doc/howto/enum.rst [hunk 1]
```diff
@@ -1353,6 +1353,13 @@ to handle any extra arguments::
     members; it is then replaced by Enum's :meth:`__new__` which is used after
     class creation for lookup of existing members.
 
+.. warning::
+
+    *Do not* call `super().__new__()`, as the lookup-only `__new__` is the one
```
Comment (ethanfurman) [inline Doc/howto/enum.rst]:
```suggestion
    *Do not* call ``super().__new__()``, as the lookup-only ``__new__`` is the one
```

File: Lib/enum.py [hunk 2]
```diff
@@ -1112,6 +1114,11 @@ def __new__(cls, value):
             for member in cls._member_map_.values():
                 if member._value_ == value:
                     return member
+        # still not found -- verify that members exist, in-case somebody got here mistakenly
+        # (such as via super when trying to override __new__)
+        if not cls._member_map_:
+            raise TypeError("%r has no members defined" % cls)
+        #
```
Comment (tyronexj) [inline Lib/enum.py:1121]: Is it ok to move `_member_map_` check after `_missing_` hook?
I have a derived class has no members and hook `_missing_`, it raise exception here.
@ethanfurman

Comment (ethanfurman) [reply Lib/enum.py:1121]: Unfortunately, no.  I tried to do that, but then the bug wasn't fixed.

What does your `missing()` hook do?

If you have no other alternative, you'll need to subclass `EnumType` and override the `__call__` method.

Comment (tyronexj) [reply Lib/enum.py:1121]: Thanks for the quick reply. here is an example:
```
from enum import Enum

class DynamicEnum(Enum):
    @classmethod
    def _missing_(cls, value):
        temp = object.__new__(cls)
        temp._name_ = f"{cls.__name__}_{value}"
        temp._value_ = value
        return temp

dynamic_enum = DynamicEnum(3)
print(dynamic_enum)
print(dynamic_enum.value)
```
which prints
```
DynamicEnum.DynamicEnum_3
3
```
before this change.

As my understanding, hook `_missing_` handle the missing input, a dynamic value in the above case.
so I raise the question: hook `_missing_` before empty check.

Comment (mikrodust-henrikp) [reply Lib/enum.py:1121]: I'll chime in here as well. I have the same question and got hit by this change as well. We have a setup where we do like:

```
from enum import Enum

class Result(Enum):
    @classmethod
    def _missing_(cls, value):
        v = None
        for subclass in cls.__subclasses__():
            try:
                v = subclass(value)
            except:
                pass

        return v


class CommonResult(Result):
    OK = 0
    ERR_FOO = -0x1
    ERR_BAR = -0x2

```

Previously, Result(0) returned CommonResult.OK. Now throws. It was a nifty pattern to use, and it sort of breaks the combination of using the `_missing_` and subclasses of Enum.

**Pull request merged**